### ISSUE TITLE
takes too long to open

### ISSUE THREAD

**Issue by LeiYangGH** [kalintsenkov/BlazorShop#7]
Hi，

I like the web site and i'm trying to understand it. I know C# well but not asp.net.
I've deployed the website on my personal server, you can have a try:

http://leiyang.icu:5000/

it takes about 7 seconds(and even 1 min on mobile device) to open, during which we can only see the 'ajax-loader.gif'.

This exceeds my imagination of .net core and blazor.

Could you tell me what's going on during that loading time? 

I can come up with a few possibilities:

- my method of starting server(same as readme) is not optimized --- if so, any way to make it release version?
- the database initializing --- if so, is there any way to skip the db initilizing temporialy, just let the user see homepage as soon as possible?
- blazor itself needs the .net core runtime download --- is it true?

Do you have a web page that hosts this code(as a live demo)?

Thanks!

**Comment by kalintsenkov** [comment]
1. You should deploy release version
2. You can skip the db initialization if you remove .Initialize() method from Startup class in Web.Server project

```
public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
            => app
                .UseExceptionHandling(env)
                .UseValidationExceptionHandler()
                .UseHttpsRedirection()
                .UseBlazorFrameworkFiles()
                .UseStaticFiles()
                .UseRouting()
                .UseAuthentication()
                .UseAuthorization()
                .UseEndpoints()
                .Initialize(); // this one
```
3. That's true.
4. I don't have live demo, but I will add it in future. 🙂

- Where did you deploy the app? Make sure database and web app are deployed in same region.

**Comment by LeiYangGH** [comment]
Hi @kalintsenkov 
I deployed in China Ali cloud. Db and web is in the same server.
I use mysql so i modified many db types and i found mysql ef lib cannot set auto increase id correctly, so my db actually cannot correctly setup.

I'll try your ways to use release version as well as skip db init. and feedback.

**Comment by harirajrv** [comment]
The latest version code is not working. 

Commented the .Initialize();

Only the loading gif is showing, tried Release, and Debug modes with VS.

**Comment by LeiYangGH** [comment]
too long no response. so the test env metioned above might be gone.

**Comment by harirajrv** [comment]
I took the repository into a new folder then it worked well, this time I used the console window to run it as mentioned in the ReadMe

And I was doing some changes in the CSS and razor HTML files, also I saw the changes I did, it was working normally.

After a while today it's stopped working again, now it's showing only the loading image.

I didn't do anything specific to the environment or C# code.
